powershell/VstsTaskSdk/InputFunctions.ps1

```diff
@@ -158,60 +160,121 @@ function Get-EnvVariable {
         }
     }
 
-    if ($item -and $item.value) {
-        $value = $item.value
-        Write-Verbose "$($path): '$value'"
+    Get-FinalValue -Value $item.Value @PSBoundParameters
+}
+
+function Get-SecureInput {
+    [CmdletBinding(DefaultParameterSetName = 'Require')]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Name,
+        [Parameter(Mandatory = $true)]
+        [string]$Path,
+        [Parameter(ParameterSetName = 'Require')]
+        [switch]$Require,
+        [Parameter(ParameterSetName = 'Default')]
+        [object]$Default,
+        [switch]$AsBool,
+        [switch]$AsInt)
+
+    # Attempt to get the secure variable.
+    $value = $null
+    if ($psCredential = $script:secureInputs[$Path]) {
+        $value = $psCredential.GetNetworkCredential().Password
+    } elseif (!$script:nonInteractive) {
+        # The value wasn't found. Prompt for the value if running in interactive dev mode.
```

Comment:
Interactive dev mode will still actually work nearly the same. With the small caveat that if the developer want to clear one of the values between interactive calls to "Invoke-TaskScript", then they need to un-import the VstsTaskSdk module and re-import it in order to clear the module-scoped dictionary.

For interactive dev mode, if the developer updates a value, then it will just overwrite the existing value in the internal module-scoped dictionary.